keras_contrib/layers/normalization.py

```diff
@@ -1,3 +1,235 @@
+    """
+
+    def __init__(self, epsilon=1e-3, mode=0, axis=-1, momentum=0.99,
+                 r_max_val=3., d_max_val=5., weights=None, beta_init='zero',
+                 gamma_init='one', gamma_regularizer=None, beta_regularizer=None,
+                 **kwargs):
+        self.supports_masking = True
+        self.beta_init = initializations.get(beta_init)
+        self.gamma_init = initializations.get(gamma_init)
+        self.epsilon = epsilon
+        self.mode = mode
+        self.axis = axis
+        self.momentum = momentum
+        self.gamma_regularizer = regularizers.get(gamma_regularizer)
+        self.beta_regularizer = regularizers.get(beta_regularizer)
+        self.initial_weights = weights
+        self.r_max_value = r_max_val
+        self.d_max_value = d_max_val
+        if self.mode == 0:
+            self.uses_learning_phase = True
+        super(BatchRenormalization, self).__init__(**kwargs)
+
+    def build(self, input_shape):
+        self.input_spec = [InputSpec(shape=input_shape)]
+        shape = (input_shape[self.axis],)
+
+        self.gamma = self.add_weight(shape,
+                                     initializer=self.gamma_init,
+                                     regularizer=self.gamma_regularizer,
+                                     name='{}_gamma'.format(self.name))
+        self.beta = self.add_weight(shape,
+                                    initializer=self.beta_init,
+                                    regularizer=self.beta_regularizer,
+                                    name='{}_beta'.format(self.name))
+        self.running_mean = self.add_weight(shape, initializer='zero',
+                                            name='{}_running_mean'.format(self.name),
+                                            trainable=False)
+        self.running_std = self.add_weight(shape, initializer='one',
+                                           name='{}_running_std'.format(self.name),
+                                           trainable=False)
+
+        self.r_max = K.variable(np.ones((1,)), name='{}_r_max'.format(self.name))
+
+        self.d_max = K.variable(np.zeros((1,)), name='{}_d_max'.format(self.name))
+
+        self.t = K.variable(np.zeros((1,)), name='{}_t'.format(self.name))
+
+        if self.initial_weights is not None:
+            self.set_weights(self.initial_weights)
+            del self.initial_weights
+        self.built = True
+
+    def call(self, x, mask=None):
+        if self.mode == 0 or self.mode == 2:
+            assert self.built, 'Layer must be built before being called'
+            input_shape = K.int_shape(x)
+
+            reduction_axes = list(range(len(input_shape)))
+            del reduction_axes[self.axis]
+            broadcast_shape = [1] * len(input_shape)
+            broadcast_shape[self.axis] = input_shape[self.axis]
+
+            if K.backend() == 'tf':
```

Comment:
Ok, because it might be generally useful, you could add a `moments` method to both the backends and then call that here as `mean_batch, var = KC.moments(x, reduction_axes)`